File: nosh_now_application/lib/data/models/location.dart
import 'package:nosh_now_application/data/models/eater.dart';

class Location {
  int locationId;
  String locationName;
  String phone;
  String coordinator;
  bool defaultLocation;
  Eater? eater;
  Location(
      {required this.locationId,
      required this.locationName,
      required this.coordinator,
      required this.phone,
      required this.defaultLocation,
      this.eater});
  factory Location.fromJson(Map<String, dynamic> json) {
    return Location(
      locationId: json['id'],
      locationName: json['locationName'],
      phone: json['phone'],
      coordinator: json['coordinator'],
      defaultLocation: json['defaultLocation'],
      eater: json['eater'] != null ? Eater.fromJson(json['eater']) : null,
    );
  }
}

File: nosh_now_apis/Models/Location.cs
namespace MyApp.Models
{
    public class Location : IModel
    {
        public int Id { get; set;}
        public string Phone { get; set;}
        public string LocationName { get; set;}
        public string Coordinator { get; set;}
        public bool Default { get; set;}    
        public int EaterId { get; set;}  
        public virtual Eater Eater{ get; set;}
    }
}